indexer/src/types/transaction.ts

```diff
@@ -73,13 +73,14 @@ export function toEthTx({
   }
   // If the transaction is a legacy, we can calculate it from the v value.
   // v = 35 + 2 * chainId + yParity -> chainId = (v - 35) / 2
-  const chainId = isLegacyTx(transaction) &&
-      transaction.supports(Capability.EIP155ReplayProtection)
-    ? bigIntToHex((BigInt(txJSON.v) - 35n) / 2n)
-    : txJSON.chainId;
+  const chainId =
+    isLegacyTx(transaction) &&
+    transaction.supports(Capability.EIP155ReplayProtection)
+      ? bigIntToHex((BigInt(txJSON.v) - 35n) / 2n)
+      : txJSON.chainId;
 
   const result: JsonRpcTx & { yParity?: string } = {
-    blockHash: isPendingBlock ? null : blockHash,
+    blockHash: isPendingBlock ? NULL_BLOCK_HASH : blockHash,
```

Comment:
this stays null for a pending transaction

Reply:
actually not really because you wouldn't want to have getTransactionByHash return a null blockhash but a valid block number 
while getTransactionReceipt for the same tx return a valid receipt

Reply:
you need to return a valid block hash for receipt and logs but not for transaction. Also it doesn't change much since we change the block hash later, but allows us to indicate to SDK's that the transaction is still pending and it's confirmed yet.
The reason for fixing the null block hash on receipt and logs was only for sdk's, nothing else.